libs/execution_context/ebpf_program.c

```diff
@@ -540,6 +549,9 @@ _ebpf_program_type_specific_program_information_attach_provider(
     if (lock_held) {
         ebpf_lock_unlock(&program->lock, state);
     }
+    if (extension_program_data) {
+        ebpf_program_data_free((ebpf_program_data_t*)extension_program_data);
+    }
```

Comment:
```suggestion
    ebpf_program_data_free((ebpf_program_data_t*)extension_program_data);
    extension_program_data = NULL;
```
No need for the IF test, which just increases code complexity and lowers code coverage numbers.